pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -4116,97 +4116,104 @@
-      !>  ^-  (unit @t)
-      =<  eauth-url:eauth:authentication
-      (per-server-event [eny *duct now rof] server-state.ax)
-    ::
-        [%authenticated %cookie @ ~]
-      ?~  cookies=(slaw %t i.t.t.tyl)  [~ ~]
-      :^  ~  ~  %noun
-      !>  ^-  ?
-      %-  =<  request-is-authenticated:authentication
-          (per-server-event [eny *duct now rof] server-state.ax)
-      %*(. *request:http header-list ['cookie' u.cookies]~)
-    ::
-        [%cache @ @ ~]
-      ?~  aeon=(slaw %ud i.t.tyl)        [~ ~]
-      ?~  url=(slaw %t i.t.t.tyl)        [~ ~]
-      ?~  entry=(~(get by cache) u.url)  [~ ~]
-      ?.  =(u.aeon aeon.u.entry)         [~ ~]
-      ?~  val=val.u.entry                [~ ~]
-      ``noun+!>(u.val)
-    ::
-        [%'_~_' *]
-      =/  mym  (scry-mime now rof (deft:de-purl:html tyl))
-      ?:  ?=(%| -.mym)  [~ ~]
-      ``noun+!>(p.mym)
+        %host
+      %-  (lift (lift |=(a=hart:eyre [%hart !>(a)])))
+      ^-  (unit (unit hart:eyre))
+      =.  p.lot  ?.(=([%da now] p.lot) p.lot [%tas %real])
+      ?+  p.lot
+        [~ ~]
+      ::
+          [%tas %fake]
+        ``[& [~ 8.443] %& /localhost]
+      ::
+          [%tas %real]
+        =*  domains  domains.server-state.ax
+        =*  ports  ports.server-state.ax
+        =/  =host:eyre  [%& ?^(domains n.domains /localhost)]
+        =/  port=(unit @ud)
+          ?.  ?=(^ secure.ports)
+            ?:(=(80 insecure.ports) ~ `insecure.ports)
+          ?:(=(443 u.secure.ports) ~ secure.ports)
+        ``[?=(^ secure.ports) port host]
+      ==
     ==
-  ?.  ?=(%$ ren)
-    [~ ~]
-  ?+  syd  [~ ~]
-    %bindings              ``noun+!>(bindings.server-state.ax)
-    %connections           ``noun+!>(connections.server-state.ax)
-    %authentication-state  ``noun+!>(auth.server-state.ax)
-    %channel-state         ``noun+!>(channel-state.server-state.ax)
-  ::
-      %host
-    %-  (lift (lift |=(a=hart:eyre [%hart !>(a)])))
-    ^-  (unit (unit hart:eyre))
-    =.  p.lot  ?.(=([%da now] p.lot) p.lot [%tas %real])
-    ?+  p.lot
-      [~ ~]
-    ::
-        [%tas %fake]
-      ``[& [~ 8.443] %& /localhost]
+  :: public endpoints
+  ?+  tyl  ~
+      [%eauth %url ~]
+    =*  endpoint  endpoint.auth.server-state.ax
+    ?.  ?=(%da -.p.lot)  [~ ~]
+    ::  we cannot answer for something prior to the last set time,
+    ::  or something beyond the present moment.
+    ::
+    ?:  ?|  (lth q.p.lot time.endpoint)
+            (gth q.p.lot now)
+        ==
+      ~
+    :^  ~  ~  %noun
+    !>  ^-  (unit @t)
+    =<  eauth-url:eauth:authentication
+    (per-server-event [eny *duct now rof] server-state.ax)
+ ::
+      [%cache @ @ ~]
+    =,  server-state.ax
+    ?~  aeon=(slaw %ud i.t.tyl)        [~ ~]
+    ?~  url=(slaw %t i.t.t.tyl)        [~ ~]
+    ?~  entry=(~(get by cache) u.url)  [~ ~]
+    ?.  =(u.aeon aeon.u.entry)         [~ ~]
+    ?~  val=val.u.entry                [~ ~]
+    ?:  &(auth.u.val !=([~ ~] lyc))    ~ 
+    ``noun+!>(u.val)
```

Comment:
Same control flow nitpick: both of these public endpoints should be available to fully-privileged requests. I don't see any simple way to do that without just moving them up into separate `?:  ?=` expressions, above all the private endpoints.

Additionally, both of these endpoints should only be defined for a `$care of `%x` and a `$desk` of `%$`. That constraint was lost in the refactoring.